data_model/src/expression.rs

```diff
@@ -373,115 +374,65 @@ mod operation {
-                    )+
-                }
-
-                impl<'re> From<&'re super::super::Expression> for $i<'re> {
-                    fn from(expression: &'re super::super::Expression) -> Self {
-                        match expression {
-                            $(
-                                super::super::Expression::$var(value) => Self::$var(&value),
-                            )+
-                        }
-                    }
-                }
-            }
-        }
-    };
-}
-
-expression_serde_internal_repr! {
-    /// Represents all possible expressions.
-    #[derive(
-        Debug, Display, Clone, PartialEq, Eq, Hash, Decode, Encode, FromVariant, IntoSchema, PartialOrd, Ord,
-    )]
-    pub enum Expression {
-        /// Add expression.
-        Add(Add),
-        /// Subtract expression.
-        Subtract(Subtract),
-        /// Multiply expression.
-        Multiply(Multiply),
-        /// Divide expression.
-        Divide(Divide),
-        /// Module expression.
-        Mod(Mod),
-        /// Raise to power expression.
-        RaiseTo(RaiseTo),
-        /// Greater expression.
-        Greater(Greater),
-        /// Less expression.
-        Less(Less),
-        /// Equal expression.
-        Equal(Equal),
-        /// Not expression.
-        Not(Not),
-        /// And expression.
-        And(And),
-        /// Or expression.
-        Or(Or),
-        /// If expression.
-        If(If),
-        /// Raw value.
-        Raw(ValueBox),
-        /// Query to Iroha state.
-        Query(QueryBox),
-        /// Contains expression for vectors.
-        Contains(Contains),
-        /// Contains all expression for vectors.
-        ContainsAll(ContainsAll),
-        /// Contains any expression for vectors.
-        ContainsAny(ContainsAny),
-        /// Where expression to supply temporary values to local context.
-        Where(Where),
-        /// Get a temporary value by name
-        ContextValue(ContextValue),
-    }
+/// Represents all possible expressions.
+#[derive(
+    Debug,
+    Display,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Decode,
+    Encode,
+    FromVariant,
+    IntoSchema,
+    PartialOrd,
+    Ord,
+    PartiallyTaggedSerialize,
+    PartiallyTaggedDeserialize,
+)]
+pub enum Expression {
+    /// Add expression.
+    Add(Add),
+    /// Subtract expression.
+    Subtract(Subtract),
+    /// Multiply expression.
+    Multiply(Multiply),
+    /// Divide expression.
+    Divide(Divide),
+    /// Module expression.
+    Mod(Mod),
+    /// Raise to power expression.
+    RaiseTo(RaiseTo),
+    /// Greater expression.
+    Greater(Greater),
+    /// Less expression.
+    Less(Less),
+    /// Equal expression.
+    Equal(Equal),
+    /// Not expression.
+    Not(Not),
+    /// And expression.
+    And(And),
+    /// Or expression.
+    Or(Or),
+    /// If expression.
+    If(If),
+    /// Raw value.
+    #[serde_partially_tagged(untagged)]
+    Raw(ValueBox),
```

Comment:
AFAICT the new proc macro is a complete replacement for the declarative macro we had here, and also more general? Good news again.

Reply:
> AFAICT the new proc macro is a complete replacement for the declarative macro we had here, and also more general?

Yes